src/manage_nsfs/nc_lifecycle.js

```diff
@@ -47,6 +47,12 @@ const TIMED_OPS = Object.freeze({
     DELETE_MULTIPLE_OBJECTS: 'delete_multiple_objects'
 });
 
+/**
+ * @typedef {{
+ * is_finished?: Boolean | Undefined,
+ * expire?: { is_finished?: Boolean | Undefined, key_marker?: String | Undefined, candidates_file_offset?: number | undefined}
+ * noncurrent?: { is_finished?: Boolean | Undefined, key_marker_versioned?: String | Undefined, version_id_marker?: String | Undefined }}} RuleState
+*/
```

Comment:
minor suggestion - add a newline
```suggestion
 * noncurrent?: { is_finished?: Boolean | Undefined, key_marker_versioned?: String | Undefined, version_id_marker?: String | Undefined }
 }} RuleState
*/
```